088-rebalance-progress-status.md

```diff
@@ -0,0 +1,374 @@
+- [1] The maximum number of partition movements given Cruise Control partition movement capacity
+- [2] We don't have a method of determining disk throughput.
+- [3] The network bandwidth given Cruise Control bandwidth throttle.
+- [4] The throughput given the max allowed number of partition movements and disk throughput.
+ 
+#### State: `Rebalancing`
+
+$$
+\text{rate} = \frac{\text{finishedDataMovement}\_{[1]}}{\text{taskTriggerTime}\_{[2]} - \text{currentTime}}
+$$
+
+$$
+\text{estimatedTimeToCompletionInMinutes} = \frac{\text{totalDataToMove}\_{[3]} - \text{finishedDataMovement}\_{[1]}}{\text{rate}}
+$$
+
+Notes
+- [1] The number of megabytes already moved by rebalance, provided by [/kafkacruisecontrol/state?substates=executor](#field-executorstate) REST API endpoint.
+- [2] The time when the rebalance task was started, extracted from `triggeredTaskReason` field from the [/kafkacruisecontrol/state?substates=executor](#field-executorstate) for that task.
+- [3] The total number of megabytes planned to be moved for rebalance, provided from json payload of the [/kafkacruisecontrol/state?substates=executor](#field-executorstate) REST API endpoint.
+
+#### State: `Stopped`
+
+$$
+\text{estimatedTimeToCompletionInMinutes} = \text{N/A}
+$$
+
+Once a rebalance has been stopped, it cannot be resumed. 
+Therefore, there is no `estimatedTimeToCompletionInMinutes` for a stopped rebalance. We set the field to `N/A` to emphasize this. 
+To move from the `Stopped` state, a user must refresh the `KafkaRebalance` resource, the `rebalanceProgressConfigMap` will then be updated by the operator upon the next state change.
+
+#### State: `Ready`
+
+$$
+\text{estimatedTimeToCompletionInMinutes} = 0
+$$
+
+The rebalance is complete so we hardcode the value to `0`
+This emphasizes that the rebalance is complete and helps clear up ambiguity surrounding what the `Ready` state means in the `KafkaRebalance` resource.
+
+### Field: `completedDataMovementPercentage`
+
+The percentage of the data movement of the partition rebalance that is completed.
+
+The formulas used to calculate field value per  `KafkaRebalance` state:
+
+#### State: `ProposalReady`
+
+$$
+\text{completedDataMovementPercentage} = 0\%
+$$
+
+The rebalance has not started yet so no data has been moved, so we hardcode the value to `0`.
+This emphasizes that the rebalance has not started and helps clear up ambiguity surrounding what the `optimizationResult` field means in the `KafkaRebalance` resource.
+
+#### State: `Rebalancing`
+
+$$
+\text{completedDataMovementPercentage} = (\frac{\text{finishedDataMovement}\_{[1]}}{\text{totalDataToMove}\_{[2]}} \times 100)\%
+$$
+
+Notes
+ - [1] The number of megabytes already moved by rebalance, provided by [/kafkacruisecontrol/state?substates=executor](#field-executorstate) REST API endpoint.
+ - [2] The total number of megabytes planned to be moved for rebalance, provided by [/kafkacruisecontrol/state?substates=executor](#field-executorstate) REST API endpoint.
```

Comment:
Can we put the specific fields we are using from the returned JSON for both the above?